server/src/platform/drbg/mod.rs

```diff
@@ -0,0 +1,128 @@
+/// with the [`next_u32`] and [`next_u64`] methods, implementations may
+/// discard some random bits for efficiency.
+/// Implementation should normally have portable, reproducible output,
+/// i.e. fix Endianness when converting values to avoid platform
+/// differences, and avoid making any changes which affect output.
+///
+/// A DRBG may implement only one of the [`next_u32`], [`next_u64`],
+/// [`fill_bytes`] methods available in this trait directly, then use
+/// the helper functions from the [helpers] module to implement the
+/// other methods.
+///
+/// ## Implementing [`Drbg`] for large entropy or nonce
+///
+/// Note that the required traits [`core::default::Default`] and
+/// [`core::convert::AsMut<u8>`] are not implemented for large arrays
+/// [u8; N] with N > 32. To be able to implement the traits required
+/// by SeedableRng for RNGs with such large seeds, the newtype pattern
+/// can be used:
+///
+/// ```rust
+/// const N: usize = 64;
+/// pub struct MyRngSeed(pub [u8; N]);
+///
+/// impl Default for MyRngSeed {
+///     fn default() -> MyRngSeed {
+///         MyRngSeed([0; N])
+///     }
+/// }
+///
+/// impl AsMut<[u8]> for MyRngSeed {
+///     fn as_mut(&mut self) -> &mut [u8] {
+///         &mut self.0
+///     }
+/// }
+/// ```
+///
+/// [`fill_bytes`]: Drbg::fill_bytes
+/// [`next_u32`]: Drbg::next_u32
+/// [`next_u64`]: Drbg::next_u64
+pub trait Drbg: Sized {
+    type Error: Debug;
+    /// Entropy type, which is restricted to types mutably-dereferenceable as u8 arrays
+    /// (we recommend either `[u8; N]` for some `N` or `GenericArray<u8, N>`).
+    type Entropy: Sized + Default + AsMut<[u8]>;
+    /// Nonce type, which is restricted to types mutably-dereferenceable as u8 arrays
+    /// (we recommend either `[u8; N]` for some `N` or `GenericArray<u8, N>`).
+    ///
+    /// According to 8.6.7: A nonce may be composed of one (or more) of the following
+    /// components (other components may also be appropriate):
+    ///
+    /// 1. A random value that is generated anew for each nonce, using an approved random bit
+    ///    generator.
+    /// 2. A timestamp of sufficient resolution (detail) so that it is different each time it
+    ///    is used.
+    /// 3. A monotonically increasing sequence number.
+    /// 4. A combination of a timestamp and a monotonically increasing sequence number, such
+    ///    that the sequence number is reset when and only when the timestamp changes. For
+    ///    example, a timestamp may show the date but not the time of day, so a sequence
+    ///    number is appended that will not repeat during a particular day.
+    ///
+    /// For case 1 above, the random value could be acquired from the same source and at the same time
+    /// as the entropy input. For case 2 above, the timestamp must be trusted. A trusted timestamp is
+    /// generated and signed by an entity that is trusted to provide accurate time information.
+    type Nonce: Sized + Default + AsMut<[u8]>;
```

Comment:
This is only used for `instantiate`, which as I've mentioned below feels like something that should not be a trait function. If we in fact remove `instantiate`, we should probably remove this too.